core/process.py

```diff
@@ -548,9 +545,11 @@ def Push(self, redirects, err_out):
             #log('apply %s', r)
             with ui.ctx_Location(self.errfmt, r.op_loc):
                 try:
-                    self._ApplyRedirect(r)
+                    self._ApplyRedirect(r, err_out)
                 except (IOError, OSError) as e:
-                    err_out.append(e)
+                    err_out.append(e.errno)
```

Comment:
I still have to keep the try/except block as some calls inside `_ApplyRedirect` still raise `IOError` and `OSError`.

TODO: Figure out what these are and catch inside `_ApplyRedirect`?